Library/include/CSP/CSPFoundation.h

```diff
@@ -55,6 +55,9 @@ class CSP_API ClientUserAgent
 
 	// @brief Magnopus services environment. e.g odev.
 	csp::common::String CHSEnvironment;
+
+	// @brief Maintainance Window Url.
+	csp::common::String MaintainanceWindowURI;
```

Comment:
The intent with this struct is that it is used to derive the useragent header we sent with HTTP requests. I can see why we went with this as a home for the maintenance window URI, but perhaps `EndpointURIs` would be a better home?

Reply:
That was my suggestion (clientagentinfo) as an alternative to adding it as an argument to the initialise call, but I agree  EndpointURIs would be better still.

Reply:
Yes I agree

Reply:
Ok, changes incoming soon then

Reply:
Appreciate that's a breaking change. I'd be fine with it though given the nature of the request.

Reply:
Either one is a breaking change, as both APIs are public facing

Reply:
That's fair, and since my original suggestion of the endpoint route seems non-viable (due to it being immutable for CSP client devs), I guess we go with the maintenance system approach.

Reply:
I've made the suggested rework changes and are now ready for review

Reply:
Love it